src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -748,8 +751,11 @@ public void run() {
 					GetRecordsRequest getRecordsRequest = new GetRecordsRequest();
 					getRecordsRequest.setShardIterator(ShardConsumer.this.shardIterator);
 					getRecordsRequest.setLimit(KinesisMessageDrivenChannelAdapter.this.recordsLimit);
-					GetRecordsResult result = KinesisMessageDrivenChannelAdapter.this.amazonKinesis
-							.getRecords(getRecordsRequest);
+					GetRecordsResult result = getRecords(getRecordsRequest);
+					if (result == null) {
+						ShardConsumer.this.task = null;
+						return;
```

Comment:
I think that looks more cleaner to have an opposite `if (result != null) ` and wrap to it the next operations.